stat.go

```diff
@@ -648,6 +648,47 @@ func KullbackLeibler(p, q []float64) float64 {
 	return kl
 }
 
+// OrdinaryLinearRegression computes the simple linear regression of x and y.
+// It returns the values for \hat{\alpha}, \hat{\beta} and the coefficient
+// of determination, r. If origin is true, the regression is forced to pass
+// through the origin.
+func OrdinaryLinearRegression(x, y, weights []float64, origin bool) (alpha, beta, r float64) {
+	w := 1.0
+	yMean := Mean(y, weights)
+	if origin {
+		var x2Sum, xySum float64
+		for i, xi := range x {
+			if weights != nil {
```

Comment:
perhaps a premature optimization, but...
instead of doing this test over and over here (and later, at line 677), wouldn'it be better to create a small `func` literal:

```go
weight := func(i int) float64 {
    return weights[i]
}
if weights == nil {
    weight = func(int) float64 { return 1 }
}
```
and then use that everywhere ?